File: backend/app/schemas/triage.py
from pydantic import BaseModel, Field
from typing import Optional, List
from app.utils.enums import Priority, RecommendedAction, AgeGroup, ConfidenceLevel
from app.schemas.vital import VitalInput

class ReasonItem(BaseModel):
    code: str
    message: str

class TriageRequest(BaseModel):
    patientId: Optional[str] = None  # If provided, update existing patient
    name: Optional[str] = None
    age: int = Field(..., ge=0, le=150)
    gender: str
    chiefComplaint: str
    symptoms: List[str] = []
    historyAvailable: bool = True
    medicalHistory: Optional[List[str]] = None
    vitals: VitalInput

class TriageResponse(BaseModel):
    patientId: str
    priority: Priority
    riskScore: float
    confidence: float
    confidenceLevel: ConfidenceLevel
    safetyFloor: Optional[Priority]
    safetyFlags: List[str]
    reasons: List[ReasonItem]
    recommendedAction: RecommendedAction
    ageGroup: AgeGroup
    dataQuality: float
    limitations: List[str]
    modelVersion: str
    deteriorating: bool
    deteriorationSeverity: Optional[str]
    explanation: Optional[str] = None

File: frontend/types/triage.ts
import { Priority, RecommendedAction, ReasonItem, AgeGroup } from './patient';

export interface TriageRequest {
  patientId?: string;
  name?: string;
  age: number;
  gender: string;
  chiefComplaint: string;
  symptoms: string[];
  historyAvailable: boolean;
  medicalHistory?: string[];
  vitals: {
    heartRate?: number;
    systolicBp?: number;
    diastolicBp?: number;
    spo2?: number;
    temperature?: number;
    respiratoryRate?: number;
  };
}

export interface TriageResponse {
  patientId: string;
  priority: Priority;
  riskScore: number;
  confidence: number;
  confidenceLevel: 'HIGH' | 'MODERATE' | 'LOW';
  safetyFloor: Priority | null;
  safetyFlags: string[];
  reasons: ReasonItem[];
  recommendedAction: RecommendedAction;
  ageGroup: AgeGroup;
  dataQuality: number;
  limitations: string[];
  modelVersion: string;
  deteriorating: boolean;
  deteriorationSeverity: string | null;
}
